File: programmers/12910/solution.go
package main

import (
    "sort"
)

func solution(arr []int, divisor int) []int {
    var data []int

    for _, num := range arr {
        if num%divisor == 0 {
            index := sort.Search(len(data), func(i int) bool { return data[i] > num })
            data = append(data, num)
            copy(data[index+1:], data[index:])
            data[index] = num
        }
    }

    if data == nil {
        data = append(data, -1)
    }

    return data
}

File: programmers/12910/solution.kt
class Solution {
    fun solution(arr: IntArray, divisor: Int): IntArray {
        var answer = ArrayList<Int>()
        var idx: Int = 0

        for (e in arr) {
            if (e%divisor == 0) {
                answer.add(e)
            }
        }
        if (answer.isEmpty()) answer.add(-1)

        return answer.toIntArray().sortedArray()
    }
}
